File: backend/src/apps/portfolio/serializers/hobbies.py
from rest_framework import serializers

from src.apps.portfolio.models import Hobby, HobbyTag


class HobbyTagSerializer(serializers.ModelSerializer):
    class Meta:
        model = HobbyTag
        fields = "__all__"


class HobbyListSerializer(serializers.ModelSerializer):
    class Meta:
        model = Hobby
        fields = (
            "id",
            "user",
            "name",
            "description",
            "started_at",
            "ended_at",
            "image",
        )


class HobbyRetrieveSerializer(serializers.ModelSerializer):
    tags = HobbyTagSerializer(many=True)

    class Meta:
        model = Hobby
        fields = (
            "id",
            "user",
            "name",
            "description",
            "started_at",
            "ended_at",
            "image",
            "tags",
        )

File: frontend/src/types/hobbies.ts
export type HobbyTag = {
  id: number;
  name: string;
  slug: string;
};

export type Hobby = {
  id: number;
  name: string;
  image: string;
  description: string;
  started_at: string;
  ended_at?: string;
  tags: HobbyTag[];
};
